tidy3d/components/geometry.py

```
@@ -1316,6 +1316,9 @@ def no_self_intersecting_polygon(cls, val, values):
         if they are self-intersecting.
         """
 
+        # is sidewal_angle a valid value?
+        if "sidewall_angle" not in values:
+            raise ValidationError("``sidewall_angle`` needs to be in the range (-np.pi/2,np.pi/2).")
```

let's just raise a generic error like "``sidewall_angle`` failed validation, can't test ``no_self_intersecting_polygon``"